File: src/main/java/com/sfh/pokeRogueBot/model/run/WavePokemon.java
package com.sfh.pokeRogueBot.model.run;

import com.google.gson.annotations.SerializedName;
import com.sfh.pokeRogueBot.model.poke.Pokemon;
import lombok.Data;

@Data
public class WavePokemon {

    @SerializedName("enemyParty")
    private Pokemon[] enemyTeam;
    @SerializedName("ownParty")
    private Pokemon[] playerTeam;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/run/WavePokemon.kt
package com.sfh.pokeRogueBot.model.run

import com.google.gson.annotations.SerializedName
import com.sfh.pokeRogueBot.model.poke.Pokemon

data class WavePokemon(
    var enemyParty: List<Pokemon>,
    @SerializedName("ownParty")
    var playerParty: List<Pokemon>
)